File: 卡码/代码随想录/1/01-数组/Js/02-双指针/doublePointer.js
//27. 移除元素
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

const removeElement = (nums, val)=> {
  let slow=0;
  for(let fast=0;fast<nums.length;fast++){
    if(nums[fast]!==val) nums[slow++]=nums[fast];
  }
  return slow;
};

async function main(){
  const nums=[0,1,2,2,3,0,4,2];
  console.log(removeElement(nums,2));
}

main();

File: 卡码/代码随想录/1/01-数组/Go/02-双指针/doublePointer.go
//27. 移除元素
package main

import "fmt"

func removeElement(nums []int, val int) int {
	fast,slow:=0,0
	for fast=0;fast<len(nums);fast++ {
		if nums[fast]!=val {
			nums[slow]=nums[fast]
			slow++
		}
	}
	return slow
}

func main(){
	// nums:=[]int{3,2,2,3}
	nums:=[]int{0,1,2,2,3,0,4,2}
	n:=removeElement(nums,2)
	fmt.Println(n)
}
